hamlet/views.py

```diff
@@ -121,7 +121,7 @@ def ml_subject_assistant_list(request, project_id):
             ).order_by('-created')
             
             external_web_app_url = settings.SUBJECT_ASSISTANT_EXTERNAL_URL 
-            if list(data_export)[0].ml_task_id:
+            if data_export and list(data_export)[0].ml_task_id:
```

Comment:
This looks good for fixing the `IndexError`, but maybe it would be better to use the `.first` method (https://docs.djangoproject.com/en/3.0/ref/models/querysets/#first) on the query set instead (because then you can avoid converting all the results to a list just to get the first one):

```python
if data_export.first and data_export.first.ml_task_id:
    external_web_app_url = external_web_app_url + str(data_export.first.ml_task_id)
```

Reply:
👍 Let me try that out!